packages/editor/src/components/editor-history/redo.js

```diff
@@ -3,17 +3,20 @@
  */
 import { __ } from '@wordpress/i18n';
 import { Button } from '@wordpress/components';
-import { withSelect, withDispatch } from '@wordpress/data';
-import { compose } from '@wordpress/compose';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { displayShortcut } from '@wordpress/keycodes';
 import { redo as redoIcon } from '@wordpress/icons';
 import { forwardRef } from '@wordpress/element';
 
-function EditorHistoryRedo( { hasRedo, redo, innerRef, ...props } ) {
+function EditorHistoryRedo( props, ref ) {
+	const hasRedo = useSelect( ( select ) =>
+		select( 'core/editor' ).hasEditorRedo()
+	);
```

Comment:
Noting that you forgot to add a dependency array to this and the other `useSelect` calls to memoize the callbacks.

Reply:
Oh, thanks for noticing! I'll create a follow-up PR.

Reply:
We really need a lint rule, it’s impossible to catch all cases in review.